runtime/wasm/src/host_exports.rs

```diff
@@ -404,6 +407,22 @@ where
             .unwrap();
         return_receiver.wait().expect("`return_sender` dropped")
     }
+
+    pub(crate) fn check_timeout(
+        &self,
+        start_time: Instant,
+    ) -> Result<(), HostExportError<impl ExportError>> {
+        let event_handler_timeout = std::env::var(TIMEOUT_ENV_VAR)
+            .ok()
+            .and_then(|s| u64::from_str(&s).ok())
+            .map(Duration::from_secs);
```

Comment:
Would it make sense to parse the duration once instead of doing it for every gas metering call? I'm not sure how expensive it is, but it could slow execution down.

Reply:
I don't think reading and parsing the env var is extraordinarily expensive. Improving performance here is still a good thought, but we'd be trading off the ability to change this variable during the execution of the mapping.

Reply:
It may be a premature optimization. Let's roll with the current approach.

Reply:
@lutter Sorry for missing this. WASM modules are sandboxed and don't have access to env vars, and generally don't have access to the host environment memory or syscalls.